lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -307,6 +307,89 @@ def test_index_no_visible_section_in_chapter(self):
         self.assertNotIn('Problem 1', response.content)
         self.assertNotIn('Problem 2', response.content)
 
+    def _create_staff_user_and_course(self):
+        """
+        User will have staff access and creates the course
+        """
+        self.user = GlobalStaffFactory.create()
+        self.client.login(username=self.user.username, password='test')
+
+        # create the course
+        course = CourseFactory.create()
+        return course
+
+    def _create_url_for_enroll_staff(self, course):
+        """
+        creates the courseware url and enroll staff url
+        """
+        # create the _next parameter
+        courseware_url = reverse('courseware', kwargs={
+            'course_id': unicode(course.id)}) + '?activate_block_id=test_block_id'
```

Comment:
instead of `test_block_id`, if you use `self.course`, you can use the block id of an existing problem. So you could use `unicode(self.problem.location)`. Instead. Also, you might want to use `urlencode` to create this.